lib/python/ray/worker.py

```diff
@@ -225,6 +230,8 @@ def func_call(*args, **kwargs):
       """This is what gets run immediately when a worker calls a remote function."""
       args = list(args)
       args.extend([kwargs[keyword] if kwargs.has_key(keyword) else default for keyword, default in func_call.keyword_defaults[len(args):]]) # fill in the remaining arguments
+      if worker.mode == ray.PYTHON_MODE:
+        return func(*[copy.deep_copy(arg) for arg in args]) # In ray.PYTHON_MODE, remote calls simply execute the function. We copy the arguments to avoid mutating them.
```

Comment:
I'll add a comment, but in Ray, objects are immutable, so passing an object into a remote function can never change the object. However, that isn't the case with regular Python functions, which is why I added that. I didn't actually produce an example that caused the problem, so I'll add that as well.